pkg/openapiclient/local_schemas.go

```diff
@@ -14,55 +13,49 @@ import (
 
 // client which provides openapi read from files on disk
 type localSchemasClient struct {
-	fs  fs.FS
-	dir string
+	fs fs.FS
 }
 
 // Dir should have openapi files following directory layout:
 // /<apis>/<group>/<version>.json
 // /api/<version>.json
-func NewLocalSchemaFiles(fs fs.FS, dirPath string) openapi.Client {
+func NewLocalSchemaFiles(fs fs.FS) openapi.Client {
 	return &localSchemasClient{
-		fs:  fs,
-		dir: dirPath,
+		fs: fs,
 	}
 }
 
 func (k *localSchemasClient) Paths() (map[string]openapi.GroupVersion, error) {
-	if len(k.dir) == 0 {
+	if k.fs == nil {
 		return nil, nil
 	}
 	res := map[string]openapi.GroupVersion{}
-	apiGroups, err := utils.ReadDir(k.fs, filepath.Join(k.dir, "apis"))
-	if err != nil && !errors.Is(err, fs.ErrNotExist) {
-		return nil, fmt.Errorf("failed reading local files dir %s: %w", filepath.Join(k.dir, "apis"), err)
-	}
-	for _, f := range apiGroups {
-		groupPath := filepath.Join(k.dir, "apis", f.Name())
-		versions, err := utils.ReadDir(k.fs, groupPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed reading local files dir %s: %w", groupPath, err)
+	if apiGroups, err := fs.ReadDir(k.fs, "apis"); err == nil {
```

Comment:
note that the behaviour changes slightly. this is because it seems impossible to have a consistent behaviour between windows and linux :(

see https://github.com/golang/go/issues/46734

Reply:
IIRC the inconsistency case was when calling `ReadDir` on a file.

> That would mean for files with the name apis we throw no error and apiGroups is empty. I think that is preferrable to the current diff which throws out all errors.

In this case `ReadDir` will fail, `err == nil` will not be true and `apiGroups` is not considered at all.
My feeling was that we were becoming more permissive with this kind of errors so it made sense to check `err == nil` and just do nothing in case there was an error.

Let me know if you want me to change the behaviour but yeah the real problem is `errors.Is(err, fs.ErrNotExist)` not being consistent across all systems (at least with an error we got from `ReadDir`).

Reply:
> Let me know if you want me to change the behaviour but yeah the real problem is errors.Is(err, fs.ErrNotExist) not being consistent across all systems (at least with an error we got from ReadDir).

Is it possible to check if the file exists without checking the ReadDir `err` result? Like stat?

```go
apiGroups, err := fs.ReadDir(k.fs, "apis")
if err != nil && CrossPlatformCheckDirExists(k.fs, "apis")  {
    return nil, fmt.Errorf("...: %w", err)
}
```

Reply:
We can probably do something like that.
Ultimately do you prefer returning an error in this case ? Or just skip the error to let the chain continue collections paths ?

Reply:
If the dir does not exist it should skip it, but if it does exist and we couldn't read it then it should be an error